electrode-ota-server-model-acquisition/src/acquisition.js

```diff
@@ -52,12 +53,12 @@ export default (options, dao, weighted, _download, manifest, logger) => {
                         };
                     }
                 }
-                const pkgAppVersion = version.coerce(pkg.appVersion, true).toString();
+                // don't coerce the targetVersion, it'll remove the range
+                const pkgAppVersion = pkg.appVersion;//version.coerce(pkg.appVersion, true).toString();
                 const paramAppVersion = version.coerce(params.appVersion, true).toString();
-
-                const isNotAvailable = pkg.packageHash == params.packageHash || !("clientUniqueId" in params)
-                    || version.gt(paramAppVersion, pkgAppVersion)
-                    || pkg.isDisabled;
+                const isTargetBinaryVersion = version.satisfies(paramAppVersion, pkgAppVersion);
+                const isNotAvailable = pkg.packageHash === params.packageHash || !("clientUniqueId" in params)
+                    || !isTargetBinaryVersion || pkg.isDisabled;
```

Comment:
@datvong-wm, I believe `isTargetBinaryVersion` should be prioritized, like:
`!isTargetBinaryVersion || (pkg.packageHash === params.packageHash || !("clientUniqueId" in params) || pkg.isDisabled)`

Or if it's not the targeted version, should we send the download  URL still?

Reply:
forgot why i said that.  let's keep the same priority as before.

Reply:
It's same as before.